sentry/src/test/java/io/sentry/SentryEnvelopeItemTest.kt

```diff
@@ -20,7 +42,141 @@ class SentryEnvelopeItemTest {
+        val changedFileReadPermission = file.setReadable(false)
+        if (changedFileReadPermission) {
+            val logger = mock<ILogger>()
+            val attachment = Attachment(file.absolutePath, "file.txt")
+
+            val item = SentryEnvelopeItem.fromAttachment(logger, attachment)
+
+            assertAttachment(attachment, byteArrayOf(), item)
+            verifyLogException<FileNotFoundException>(logger, attachment.filename)
+        } else {
+            println("Was not able to change file access permission. Skipping test.")
+        }
+    }
+
+    @Test
+    fun `fromAttachment with file SecurityManager denies read access`() {
+        val file = File(fixture.filename)
+        file.writeBytes(fixture.bytes)
+
+        val logger = mock<ILogger>()
+        val attachment = Attachment(file.absolutePath, "file.txt")
+
+        val securityManager = DenyReadFileSecurityManager(fixture.filename)
+        System.setSecurityManager(securityManager)
+
+        val item = SentryEnvelopeItem.fromAttachment(logger, attachment)
+
+        assertAttachment(attachment, byteArrayOf(), item)
+        verifyLogException<SecurityException>(logger, attachment.filename)
+
+        System.setSecurityManager(null)
+    }
+
+    @Test
+    fun `fromAttachment with both bytes and path null`() {
+        val attachment = Attachment("")
+        // Annotations prevent creating attachments with both bytes and path null.
+        // If someone ignores the annotations in Java and passes null for path
+        // or bytes, we still want our code to work properly. Instead of creating
+        // an extra test class in Java and ignoring the warnings we just use
+        // reflection instead.
+        attachment.injectForField("path", null)
+
+        val item = SentryEnvelopeItem.fromAttachment(mock(), attachment)
+
+        assertAttachment(attachment, byteArrayOf(), item)
+    }
+
     private fun createSession(): Session {
         return Session("dis", User(), "env", "rel")
     }
+
+    private fun assertAttachment(
+        attachment: Attachment,
+        expectedBytes: ByteArray,
+        actualItem: SentryEnvelopeItem
+    ) {
+        assertEquals(attachment.contentType, actualItem.header.contentType)
+        assertEquals(attachment.filename, actualItem.header.fileName)
+        assertTrue(
+            expectedBytes.contentEquals(actualItem.data),
+            "${String(expectedBytes)} is not equal to ${String(actualItem.data)}"
+        )
+    }
+
+    private inline fun <reified T : Exception> verifyLogException(logger: ILogger, filename: String) {
+        verify(logger)
+            .log(eq(SentryLevel.ERROR), any<T>(),
+                eq("Serializing attachment %s failed."), eq(filename))
+    }
+
+    private inline fun <reified T : Any> T.injectForField(name: String, value: Any?) {
+        T::class.java.getDeclaredField(name)
+            .apply { isAccessible = true }
+            .set(this, value)
+    }
```

Comment:
L: we could extract it to its own Util class and reuse it, I see this happening more often after making the usage of @NotNull and @Nullable
`AndroidOptionsInitializerTest` for example, but its another Module so we can't reuse it, just to keep in mind.

Reply:
I strongly agree. What about a project private test-util Gradle module? I used them heavily in the past and it made my life and the life of many other devs way easier.

Reply:
it already exists, its called `sentry-test-support`, it was midnight so totally forgot about it :P
this module was created afterwards, so a lot of things could be moved in there